Bottom: 0cb28281eda179c7015bf22f1d87d9d62a23252b
Top:    fcd08bd796a83c0942da19bc18f11f78ae77d939
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Configure Docker Daemon


---
diff --git a/roles/docker-common/tasks/configure.yml b/roles/docker-common/tasks/configure.yml
new file mode 100644
index 00000000..c1f016e1
--- /dev/null
+++ b/roles/docker-common/tasks/configure.yml
@@ -0,0 +1,25 @@
+---
+
+- name: Create Docker Configuration Directory
+  file:
+    path: /etc/docker
+    state: directory
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+
+- name: Configure Docker Daemon
+  template: 
+    dest: /etc/docker/daemon.json
+    src: daemon.json.j2
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+  register: docker_json
+
+- name: Restart Docker Daemon
+  service:
+    name: docker
+    enabled: yes
+    state: reloaded
+  when: docker_json.changed
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index 7ef3f40c..bc8aac33 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -8,6 +8,7 @@
 - import_tasks: sshd_settings.yml
 - import_tasks: mariadb_settings.yml
 
+- import_tasks: configure.yml
 
 
 - name: Check if baculavirt selinux module exists
diff --git a/roles/docker-common/templates/daemon.json.j2 b/roles/docker-common/templates/daemon.json.j2
new file mode 100644
index 00000000..57de7390
--- /dev/null
+++ b/roles/docker-common/templates/daemon.json.j2
@@ -0,0 +1,4 @@
+{
+  "log-driver": "json-file",
+  "log-opts": {"max-size": "10m", "max-file": "3"}
+}
